# Experiment Analysis

In [1]:
import pandas as pd

df = pd.read_csv("../data/final_experiment_dataset.csv")
final_experiment_df = df.copy()
final_experiment_df.head(5)

client_id Variation  clnt_tenure_yr  clnt_tenure_mnth  clnt_age gendr  \
0    9988021      Test             5.0              64.0      79.0     U   
1    8320017      Test            22.0             274.0      34.5     M   
2    4033851   Control            12.0             149.0      63.5     M   
3    1982004      Test             6.0              80.0      44.5     U   
4    9294070   Control             5.0              70.0      29.0     U   

   num_accts        bal  calls_6_mnth  logons_6_mnth  merged_session_id  \
0        2.0  189023.86           1.0            4.0                  1   
1        2.0   36001.90           5.0            8.0                  1   
2        2.0  142642.26           5.0            8.0                  1   
3        2.0   30231.76           1.0            4.0                  1   
4        2.0   34254.54           0.0            3.0                  1   

   converted           first_time            last_time  total_duration_sec  \
0      False  2017-04-12 16:57:27  2017-04-17 15:27:07            426580.0   
1       True  2017-04-05 13:08:06  2017-04-05 13:10:05               119.0   
2       True  2017-04-05 12:04:00  2017-04-05 12:29:03              1503.0   
3       True  2017-04-17 11:59:42  2017-04-17 12:02:42               180.0   
4      False  2017-04-17 11:16:13  2017-04-17 11:16:28                15.0   

   n_events  n_errors  any_anomaly  
0        15         2        False  
1         5         0        False  
2        15         3         True  
3         5         0        False  
4         2         0        False

In [2]:
final_experiment_df.shape

(56252, 18)

### We will simply check whether the two groups are separated with bias

In [3]:
final_experiment_df["Variation"].value_counts()

Variation
Test       29970
Control    26282
Name: count, dtype: int64

In [6]:
final_experiment_df.groupby("Variation")[[
    "clnt_age",
    "clnt_tenure_yr",
    "num_accts",
    "calls_6_mnth",
    "logons_6_mnth"
]].mean()


clnt_age  clnt_tenure_yr  num_accts  calls_6_mnth  logons_6_mnth
Variation                                                                   
Control    47.819667       12.151168   2.266875      3.200099       6.236664
Test       47.652886       12.092659   2.257424      3.139206       6.178545

In [7]:
# Baseline demographics and engagement metrics are well balanced between the control and test groups, indicating successful random assignment and low risk of selection bias.

In [8]:
from scipy.stats import ttest_ind

ctrl = final_experiment_df[final_experiment_df["Variation"] == "Control"]["clnt_age"]
test = final_experiment_df[final_experiment_df["Variation"] == "Test"]["clnt_age"]

ttest_ind(ctrl, test, nan_policy="omit")


TtestResult(statistic=np.float64(1.2676830202856066), pvalue=np.float64(0.20491641833028748), df=np.float64(56250.0))

p-value ≈ 0.205
This is well above common thresholds (0.05 or 0.01).

Interpretation
We fail to reject the null hypothesis that the mean age is the same in Control and Test.

What that means here:
There is no statistically significant age difference between the two groups.
That supports good randomization and low selection bias.